actions/workflows/full_backup.yaml

```diff
@@ -1,56 +1,59 @@
-version: '2.0'
+---
+version: '1.0'
+description: Workflow that backs up the StackStorm MongoDB and Postgres databases
+input:
+  - path
+  - mistral_config
+  - mongodb_admin_username
+  - mongodb_admin_password
+  - retention_days
 
-backups.full_backup:
-  description: "Workflow that backs up the StackStorm MongoDB and Postgres databases"
-  type: direct
-  input:
-    - path
-    - mistral_config
-    - mongodb_admin_username
-    - mongodb_admin_password
-    - retention_days
+output:
+  - mongodb_dump_archive: '{{ ctx().mongodb_dump_archive }}'
+  - postgres_dump_archive: '{{ ctx().postgres_dump_archive }}'
 
-  output:
-    mongodb_dump_archive: "{{ _.mongodb_dump_archive }}"
-    postgres_dump_archive: "{{ _.postgres_dump_archive }}"
+tasks:
+  date:
+    action: core.local
+    input:
+      cmd: "date +%Y%m%d_%H%M%S"
+    next:
+      - when: '{{ succeeded() }}'
+        publish:
+          - date: '{{ result().stdout }}'
+        do:
+          - mongodb_backup
+          - postgres_backup
 
-  tasks:
+  mongodb_backup:
+    action: backups.mongodb_backup
+    input:
+      path: '{{ ctx().path }}'
+      date: '{{ ctx().date }}'
+      mongodb_admin_username: '{{ ctx().mongodb_admin_username }}'
+      mongodb_admin_password: '{{ ctx().mongodb_admin_password }}'
+      retention_days: '{{ ctx().retention_days }}'
+    next:
+      - when: '{{ succeeded() or failed() }}'
+        publish:
+          - mongodb_dump_archive: '{{ result().output.mongodb_dump_archive }}'
+        do:
+          - end
 
-    date:
-      action: core.local
-      input:
-        cmd: "date +%Y%m%d_%H%M%S"
-      publish:
-        date: "{{ task('date').result.stdout }}"
-      on-success:
-        - mongodb_backup
-        - postgres_backup
+  postgres_backup:
+    action: backups.postgres_backup
+    input:
+      path: '{{ ctx().path }}'
+      date: '{{ ctx().date }}'
+      mistral_config: '{{ ctx().mistral_config }}'
+      retention_days: '{{ ctx().retention_days }}'
+    next:
+      - when: '{{ succeeded() or failed() }}'
```

Comment:
The `when` part of this is unnecessary, the full `next` block can just be:

```yaml
next:
  - publish:
      - postgres_dump_archive: '{{ result().output.postgres_dump_archive }}'
    do:
      - end
```

Reply:
I implemented this in https://github.com/EncoreTechnologies/stackstorm-backups/pull/1 - if you merge that into your branch, it will update this PR and it should hopefully automatically resolve this change request.